packages/web-app-files/src/views/shares/SharedViaLink.vue

```diff
@@ -134,6 +138,14 @@ export default defineComponent({
   async created() {
     await this.loadResourcesTask.perform()
     this.scrollToResourceFromRoute(this.paginatedResources)
+
+    useMutationSubscription(['Files/UPDATE_RESOURCE_FIELD'], async (mutation) => {
+      if (mutation.payload.field === 'shareTypes') {
+        const keepSelectedResources = this.selectedResourcesIds
+        await this.loadResourcesTask.perform()
+        this.selectedResourcesIds = keepSelectedResources
+      }
+    })
```

Comment:
I've had completely wrong hypothesis about the whole issue. Yes, you are definitely correct, I have moved it back to setup.
The issue why it couldn't keep selected files sometimes  is because those ids sometimes change. Thus it always set selected ids, but those ids don't exist anymore. 

I fixed it by getting selected id, matching it with object in paginatedResources, grabbing file id from object, then refreshing, again matching based on file id and after that I can finally get the id and set it as selected.

Reply:
Done